### PR TITLE
DAH-2269 only show table if not loading

### PR DESCRIPTION
**Pull request by tallulahkay**
DAH-2269

I opted to not show the table until it's done loading - I thought this looked best, and Yindi approved.

Screenshot 2024-05-06 at 11 20 29 AM

### CODE REVIEW

**Comment by cliu02** [comment]
@tallulahkay Is this ticket intended to include [the work](https://sfgovdt.jira.com/browse/DAH-2269?focusedCommentId=79134) to move `No result, try adjusting your filter` up towards the header row?

**Comment by tallulahkay** [comment]
> @tallulahkay Is this ticket intended to include [the work](https://sfgovdt.jira.com/browse/DAH-2269?focusedCommentId=79134) to move `No result, try adjusting your filter` up towards the header row?

ah, thanks T! I missed that comment

File: app/javascript/components/lease_ups/LeaseUpApplicationsTableContainer.js [hunk 1]
```diff
@@ -73,18 +73,20 @@ const LeaseUpTableContainer = ({
...
-        bulkCheckboxesState={bulkCheckboxesState}
-        onBulkCheckboxClick={onBulkCheckboxClick}
-      />
+      {!loading && (
+        <LeaseUpApplicationsTable
+          dataSet={map(applications, buildRowData)}
+          listingId={listingId}
+          listingType={listingType}
+          onLeaseUpStatusChange={onLeaseUpStatusChange}
+          loading={loading}
```
Comment (jimlin-sfgov) [inline app/javascript/components/lease_ups/LeaseUpApplicationsTableContainer.js]: issue: this prop is no longer used, it should be removed.

File: spec/javascript/components/lease_ups/LeaseUpApplicationsTable.test.js [hunk 2]
```diff
@@ -0,0 +1,573 @@
...
+  a0o4U00000KIJoTQAX: false,
+  a0o4U00000KIKfEQAX: false,
+  a0o4U00000KIK3jQAH: false,
+  a0o4U00000KIXTuQAP: false,
+  a0o4U00000KIL49QAH: false,
+  a0o4U00000KIPTJQA5: false,
+  a0o4U00000KIUAsQAP: false
+}
+
+const mockDataSet = [
```
Comment (jimlin-sfgov) [inline spec/javascript/components/lease_ups/LeaseUpApplicationsTable.test.js]: suggestion (non-blocking): Move mock data to the `fixtures` directory.

File: app/javascript/components/lease_ups/LeaseUpApplicationsPage.js [hunk 3]
```diff
@@ -367,7 +376,7 @@ const LeaseUpApplicationsPage = () => {
     onSelectAllApplications: handleSelectAllApplications,
     pages: state.pages,
     preferences: listingPreferences,
-    rowsPerPage: ROWS_PER_PAGE,
+    rowsPerPage: rowsPerPage(),
```
Comment (jimlin-sfgov) [inline app/javascript/components/lease_ups/LeaseUpApplicationsPage.js]: question: How is this change from static to dynamic row count related to the loading bug?

Comment (tallulahkay) [reply app/javascript/components/lease_ups/LeaseUpApplicationsPage.js]: Part of the ticket is to move up the no data message (see comments) - the quickest way to do that is to shorten the empty view (the message shows up in the middle for react table)

Comment (jimlin-sfgov) [reply app/javascript/components/lease_ups/LeaseUpApplicationsPage.js]: Makes sense, I guess you tried updating the the `.ReactTable .rt-noData {...` entry in the stylesheet but found this solution to be easier?

Comment (tallulahkay) [reply app/javascript/components/lease_ups/LeaseUpApplicationsPage.js]: ... I did not, lemme try

Comment (tallulahkay) [reply app/javascript/components/lease_ups/LeaseUpApplicationsPage.js]: That is much better! Thanks @jimlin-sfgov !

Comment (jimlin-sfgov) [reply app/javascript/components/lease_ups/LeaseUpApplicationsPage.js]: Nice, this also fixes the issue with other kinds of tables in the Partners app

**Review by jimlin-sfgov — APPROVED** [review]
LGTM 👍

**Pull request merged**